From 496c81fc79daf03e2ce58797a3ba6dbfc059e2f6 Mon Sep 17 00:00:00 2001
From: Ziad Elhanafy <ziad.elhanafy@arm.com>
Date: Tue, 25 Jun 2024 15:49:43 +0100
Subject: [PATCH] rss/kronos: Remove LCP

Remove the related LCP code and references from the platform
code as it will not be a part of Kronos anymore.

Upstream-Status: Inappropriate [Diverges from upstream]
Signed-off-by: Ziad Elhanafy <ziad.elhanafy@arm.com>
---
 .../ext/target/arm/rss/kronos/CMakeLists.txt  |   5 -
 .../target/arm/rss/kronos/bl2/boot_hal_bl2.c  | 306 +-----------------
 .../target/arm/rss/kronos/bl2/flash_map_bl2.c |  18 --
 .../arm/rss/kronos/bl2/security_cnt_bl2.c     |   3 -
 .../ext/target/arm/rss/kronos/config.cmake    |   9 +-
 .../target/arm/rss/kronos/firmware_image_id.h |   5 +-
 .../ext/target/arm/rss/kronos/flash_layout.h  |  51 ++-
 .../rss/kronos/fw_update_agent/fwu_agent.c    |   5 +-
 .../rss/kronos/fw_update_agent/fwu_agent.h    |   1 -
 .../target/arm/rss/kronos/host_base_address.h |  31 +-
 .../ext/target/arm/rss/kronos/size_defs.h     |   1 -
 11 files changed, 33 insertions(+), 402 deletions(-)

diff --git a/platform/ext/target/arm/rss/kronos/CMakeLists.txt b/platform/ext/target/arm/rss/kronos/CMakeLists.txt
index d514bafe0..1f2d45247 100644
--- a/platform/ext/target/arm/rss/kronos/CMakeLists.txt
+++ b/platform/ext/target/arm/rss/kronos/CMakeLists.txt
@@ -166,11 +166,6 @@ target_include_directories(platform_bl2
         fw_update_agent
 )
 
-target_compile_definitions(platform_bl2
-    PRIVATE
-        RSS_LCP_COUNT=${RSS_LCP_COUNT}
-)
-
 #========================= Platform BL1_1 =====================================#
 
 target_sources(platform_bl1_1
diff --git a/platform/ext/target/arm/rss/kronos/bl2/boot_hal_bl2.c b/platform/ext/target/arm/rss/kronos/bl2/boot_hal_bl2.c
index 7326b214a..e773be0c6 100644
--- a/platform/ext/target/arm/rss/kronos/bl2/boot_hal_bl2.c
+++ b/platform/ext/target/arm/rss/kronos/bl2/boot_hal_bl2.c
@@ -10,7 +10,6 @@
 #include <string.h>
 
 #include "atu_rss_drv.h"
-#include "boot_measurement.h"
 #include "bootutil/bootutil.h"
 #include "bootutil/bootutil_log.h"
 #include "bootutil/fault_injection_hardening.h"
@@ -45,9 +44,6 @@ extern struct flash_area flash_map[];
 extern const int flash_map_entry_num;
 extern ARM_DRIVER_FLASH AP_FLASH_DEV_NAME;
 
-static uint8_t *lcp_measurement;
-static struct boot_measurement_metadata *lcp_measurement_metadata;
-
 #define ARRAY_SIZE(arr) (sizeof(arr)/sizeof((arr)[0]))
 static volatile uint8_t chip_id = 0;
 static const char * const image_names [][2] = {
@@ -57,7 +53,6 @@ static const char * const image_names [][2] = {
     {"si_cl2_primary", "si_cl2_secondary"},
     {"si_cl1_primary", "si_cl1_secondary"},
     {"si_cl0_primary", "si_cl0_secondary"},
-    {"lcp_primary", "lcp_secondary"},
     {"scp_primary", "scp_secondary"},
 };
 
@@ -164,155 +159,6 @@ static void read_chip_id()
     BOOT_LOG_INF("[BL2] Chip ID: %u", chip_id);
 }
 
-static int boot_add_data_to_shared_area(uint8_t        major_type,
-                                        uint16_t       minor_type,
-                                        size_t         size,
-                                        const uint8_t *data)
-{
-    struct shared_data_tlv_entry tlv_entry = {0};
-    struct tfm_boot_data *boot_data;
-    uintptr_t tlv_end, offset;
-
-    if (data == NULL) {
-        return -1;
-    }
-
-    boot_data = (struct tfm_boot_data *)BOOT_TFM_SHARED_DATA_BASE;
-
-    /* Check whether the shared area needs to be initialized. */
-    if ((boot_data->header.tlv_magic != SHARED_DATA_TLV_INFO_MAGIC) ||
-        (boot_data->header.tlv_tot_len > BOOT_TFM_SHARED_DATA_SIZE)) {
-
-        memset((void *)BOOT_TFM_SHARED_DATA_BASE, 0, BOOT_TFM_SHARED_DATA_SIZE);
-        boot_data->header.tlv_magic   = SHARED_DATA_TLV_INFO_MAGIC;
-        boot_data->header.tlv_tot_len = SHARED_DATA_HEADER_SIZE;
-    }
-
-    /* Get the boundaries of TLV section. */
-    tlv_end = BOOT_TFM_SHARED_DATA_BASE + boot_data->header.tlv_tot_len;
-    offset  = BOOT_TFM_SHARED_DATA_BASE + SHARED_DATA_HEADER_SIZE;
-
-    /* Check whether TLV entry is already added. Iterates over the TLV section
-     * looks for the same entry if found then returns with error.
-     */
-    while (offset < tlv_end) {
-        /* Create local copy to avoid unaligned access */
-        memcpy(&tlv_entry, (const void *)offset, SHARED_DATA_ENTRY_HEADER_SIZE);
-        if (GET_MAJOR(tlv_entry.tlv_type) == major_type &&
-            GET_MINOR(tlv_entry.tlv_type) == minor_type) {
-
-            return -1;
-        }
-
-        offset += SHARED_DATA_ENTRY_SIZE(tlv_entry.tlv_len);
-    }
-
-    /* Add TLV entry. */
-    tlv_entry.tlv_type = SET_TLV_TYPE(major_type, minor_type);
-    tlv_entry.tlv_len  = size;
-
-    /* Check integer overflow and overflow of shared data area. */
-    if (SHARED_DATA_ENTRY_SIZE(size) >
-        (UINT16_MAX - boot_data->header.tlv_tot_len)) {
-        return -1;
-    } else if ((SHARED_DATA_ENTRY_SIZE(size) + boot_data->header.tlv_tot_len) >
-               BOOT_TFM_SHARED_DATA_SIZE) {
-        return -1;
-    }
-
-    offset = tlv_end;
-    memcpy((void *)offset, &tlv_entry, SHARED_DATA_ENTRY_HEADER_SIZE);
-
-    offset += SHARED_DATA_ENTRY_HEADER_SIZE;
-    memcpy((void *)offset, data, size);
-
-    boot_data->header.tlv_tot_len += SHARED_DATA_ENTRY_SIZE(size);
-
-    return 0;
-}
-
-static int store_measurement(uint8_t index,
-                             const uint8_t *measurement,
-                             size_t measurement_size,
-                             const struct boot_measurement_metadata *metadata,
-                             bool lock_measurement)
-{
-    uint16_t minor_type;
-    uint8_t claim;
-    int rc;
-
-    minor_type = SET_MBS_MINOR(index, SW_MEASURE_METADATA);
-    rc = boot_add_data_to_shared_area(TLV_MAJOR_MBS,
-                                      minor_type,
-                                      sizeof(struct boot_measurement_metadata),
-                                      (const uint8_t *)metadata);
-    if (rc) {
-        return rc;
-    }
-
-    claim = lock_measurement ? SW_MEASURE_VALUE_NON_EXTENDABLE
-                             : SW_MEASURE_VALUE;
-    minor_type = SET_MBS_MINOR(index, claim);
-    rc = boot_add_data_to_shared_area(TLV_MAJOR_MBS,
-                                      minor_type,
-                                      measurement_size,
-                                      measurement);
-
-    return rc;
-}
-
-int boot_store_measurement(uint8_t index,
-                           const uint8_t *measurement,
-                           size_t measurement_size,
-                           const struct boot_measurement_metadata *metadata,
-                           bool lock_measurement)
-{
-    uint8_t *test_zero_measurement;
-
-    if (index >= BOOT_MEASUREMENT_SLOT_MAX) {
-        return -1;
-    }
-
-    /*
-     * LCP image requires multiple loads. If slot index is for LCP, then don't
-     * add data to the shared area.
-     */
-    if(index == BOOT_MEASUREMENT_SLOT_RT_0 + RSS_FIRMWARE_LCP_ID) {
-        if (measurement_size > PSA_HASH_LENGTH(MEASURED_BOOT_HASH_ALG)) {
-            return -1;
-        }
-
-        test_zero_measurement = calloc(measurement_size, sizeof(uint8_t));
-
-        /*
-         * If LCP measurements are not stored, store the input measurements.
-         * Else, check if the store measurements matches with the input
-         * measurements. If those doesn't match, then return -1.
-         */
-        if (memcmp(lcp_measurement, test_zero_measurement,
-                   measurement_size * sizeof(uint8_t)) == 0) {
-            memcpy(lcp_measurement, measurement,
-                   measurement_size * sizeof(uint8_t));
-            memcpy(lcp_measurement_metadata, metadata,
-                   sizeof(struct boot_measurement_metadata));
-        } else {
-            if ((memcmp(measurement, lcp_measurement,
-                        measurement_size * sizeof(uint8_t)) != 0) ||
-                (memcmp(metadata, lcp_measurement_metadata,
-                        sizeof(struct boot_measurement_metadata)) != 0)) {
-                return -1;
-            }
-        }
-        return 0;
-    }
-
-    return store_measurement(index,
-                             measurement,
-                             measurement_size,
-                             metadata,
-                             lock_measurement);
-}
-
 static int32_t gic_multiple_view_init(void)
 {
     enum atu_error_t atu_err;
@@ -945,22 +791,23 @@ static int boot_platform_post_load_scp(void)
 }
 
 /*
- * =================================== LCP ====================================
+ * ================================== AP BL2 ==================================
  */
 
-/* Fuction called before lcp firmware is loaded. */
-static int boot_platform_pre_load_lcp(void)
+/* Fuction called before ap bl2 firmware is loaded. */
+static int boot_platform_pre_load_ap_bl2(void)
 {
     enum atu_error_t atu_err;
+    uint8_t boot_index = 0;
     int32_t err;
 
-    BOOT_LOG_INF("BL2: LCP pre load start");
+    BOOT_LOG_INF("BL2: AP BL2 pre load start");
 
-    BOOT_LOG_INF("BL2: Wait for doorbell from SCP before starting LCP...");
+    BOOT_LOG_INF("BL2: Wait for doorbell from SCP before starting AP BL2...");
 
     /*
      * Wait till the SCP turns on the SYSTOP power domain. This is required
-     * as the next image that will be loaded by MCUBOOT is LCP and in order
+     * as the next image that will be loaded by MCUBOOT is AP BL2 and in order
      * to load the LCM firmware, the SYSTOP domain should be in ON state which
      * can be done by SCP only.
      */
@@ -977,137 +824,6 @@ static int boot_platform_pre_load_lcp(void)
         return 1;
     }
 
-    /* Intialize LCP boot measurements */
-    lcp_measurement = (void *)calloc(PSA_HASH_LENGTH(MEASURED_BOOT_HASH_ALG),
-                            sizeof(uint8_t));
-    lcp_measurement_metadata =
-            calloc(1, sizeof(struct boot_measurement_metadata));
-    /* Configure RSS ATU to access RSS header region for LCP0 */
-    atu_err = atu_initialize_region(&ATU_DEV_S,
-                                    HOST_LCP_IMG_HDR_ATU_ID,
-                                    HOST_LCP_HDR_ATU_BASE_S,
-                                    RSS_HDR_PHYS_BASE,
-                                    RSS_IMG_HDR_ATU_SIZE);
-    if (atu_err != ATU_ERR_NONE) {
-        return 1;
-    }
-    /*
-     * Configure RSS ATU region to access the Cluster utility space via SCP
-     * address translation window. This will require both RSS and SCP ATU to be
-     * programmed correctly.
-     */
-    atu_err = atu_initialize_region(&ATU_DEV_S,
-                                    HOST_LCP_IMG_CODE_ATU_ID,
-                                    HOST_LCP_CODE_BASE_S,
-                                    (HOST_REMOTE_CHIP_PERIPH_OFFSET(chip_id) +
-                                     HOST_LCP_0_PHYS_BASE),
-                                    HOST_LCP_ATU_SIZE);
-    if (atu_err != ATU_ERR_NONE) {
-        return 1;
-    }
-
-    BOOT_LOG_INF("BL2: LCP pre load complete");
-
-    return 0;
-}
-
-/* Fuction called after lcp firmware is loaded. */
-static int boot_platform_post_load_lcp(void)
-{
-    enum atu_error_t atu_err;
-    enum mhu_v3_x_error_t mhu_error;
-    struct boot_rsp rsp;
-    int lcp_idx, rc;
-    fih_ret fih_rc = FIH_FAILURE;
-
-    BOOT_LOG_INF("BL2: LCP post load start");
-
-    /* Clear the header from the header region but avoid removing anything that
-     * the region may be incidentally overlapping.
-     */
-    memset((void *)HOST_LCP_IMG_BASE_S, 0, BL2_HEADER_SIZE);
-
-    /* Close RSS ATU region configured to access LCP SRAM code region */
-    atu_err = atu_uninitialize_region(&ATU_DEV_S, HOST_LCP_IMG_CODE_ATU_ID);
-    if (atu_err != ATU_ERR_NONE) {
-        return 1;
-    }
-
-    /* Copy the ITCM contents from LCP0 to other LCPs */
-    for (int lcp_idx = 1; lcp_idx < RSS_LCP_COUNT; lcp_idx++) {
-        /*
-         * Configure RSS ATU region to access the SCP Cluster utility space
-         * and map to the i-th LCP's ITCM
-         */
-        atu_err = atu_initialize_region(&ATU_DEV_S,
-                                        HOST_LCP_IMG_CODE_ATU_ID,
-                                        HOST_LCP_CODE_BASE_S,
-                                        (HOST_REMOTE_CHIP_PERIPH_OFFSET(chip_id) +
-                                         HOST_LCP_N_PHYS_BASE(lcp_idx)),
-                                        HOST_LCP_ATU_SIZE);
-        if (atu_err != ATU_ERR_NONE) {
-            return 1;
-        }
-
-        /* Cleaning 'rsp' to avoid accidentally loading
-         * the NS image in case of a fault injection attack. */
-        memset((void *)&rsp, 0, sizeof(struct boot_rsp));
-
-        FIH_CALL(boot_go_for_image_id, fih_rc, &rsp, RSS_FIRMWARE_LCP_ID);
-        if (FIH_NOT_EQ(fih_rc, FIH_SUCCESS)) {
-            BOOT_LOG_ERR("Unable to find bootable LCP image");
-            FIH_PANIC;
-        }
-
-        /* Clear the header from the header region but avoid removing anything that
-         * the region may be incidentally overlapping.
-         */
-        memset((void *)HOST_LCP_IMG_BASE_S, 0, BL2_HEADER_SIZE);
-
-        /* Close RSS ATU region configured to access LCP SRAM code region */
-        atu_err = atu_uninitialize_region(&ATU_DEV_S, HOST_LCP_IMG_CODE_ATU_ID);
-        if (atu_err != ATU_ERR_NONE) {
-            return 1;
-        }
-    }
-
-    /*
-     * Save the boot measurement after LCP images are loaded.
-     */
-    rc = store_measurement(
-        (uint8_t)BOOT_MEASUREMENT_SLOT_RT_0 + RSS_FIRMWARE_LCP_ID,
-        lcp_measurement,
-        PSA_HASH_LENGTH(MEASURED_BOOT_HASH_ALG) * sizeof(uint8_t),
-        lcp_measurement_metadata, false);
-    if (rc) {
-        return -1;
-    }
-
-    /* Close RSS ATU region configured to access RSS header region for LCP */
-    atu_err = atu_uninitialize_region(&ATU_DEV_S, HOST_LCP_IMG_HDR_ATU_ID);
-    if (atu_err != ATU_ERR_NONE) {
-        return 1;
-    }
-
-    BOOT_LOG_INF("BL2: LCP post load complete");
-
-    return 0;
-}
-
-/*
- * ================================== AP BL2 ==================================
- */
-
-/* Fuction called before ap bl2 firmware is loaded. */
-static int boot_platform_pre_load_ap_bl2(void)
-{
-    enum atu_error_t atu_err;
-    int32_t result;
-    uint8_t boot_index = 0;
-    enum fwu_agent_error_t fwu_ret;
-
-    BOOT_LOG_INF("BL2: AP BL2 pre load start");
-
     /* Configure RSS ATU to access AP Secure Flash for AP BL2   */
     atu_err = atu_initialize_region(&ATU_DEV_S,
                                     RSS_ATU_AP_FLASH_ID,
@@ -1138,13 +854,11 @@ static int boot_platform_pre_load_ap_bl2(void)
         return 1;
     }
 
-    result = plat_io_storage_init(PLATFORM_GPT_IMAGE_AP);
-    if (result != 0) {
+    if (plat_io_storage_init(PLATFORM_GPT_IMAGE_AP) != 0) {
         return 1;
     }
     partition_init(PLATFORM_GPT_IMAGE_AP);
-    fwu_ret = fwu_metadata_provision_ap();
-    if (fwu_ret != FWU_AGENT_SUCCESS) {
+    if (fwu_metadata_provision_ap() != FWU_AGENT_SUCCESS) {
         return 1;
     }
 
@@ -1392,7 +1106,6 @@ static int (*boot_platform_pre_load_vector[RSS_FIRMWARE_COUNT]) (void) = {
     [RSS_FIRMWARE_SECURE_ID]        = boot_platform_pre_load_secure,
     [RSS_FIRMWARE_NON_SECURE_ID]    = boot_platform_pre_load_non_secure,
     [RSS_FIRMWARE_SCP_ID]           = boot_platform_pre_load_scp,
-    [RSS_FIRMWARE_LCP_ID]           = boot_platform_pre_load_lcp,
     [RSS_FIRMWARE_SI_CL0_ID]        = boot_platform_pre_load_si_cl0,
     [RSS_FIRMWARE_SI_CL1_ID]        = boot_platform_pre_load_si_cl1,
     [RSS_FIRMWARE_SI_CL2_ID]        = boot_platform_pre_load_si_cl2,
@@ -1404,7 +1117,6 @@ static int (*boot_platform_post_load_vector[RSS_FIRMWARE_COUNT]) (void) = {
     [RSS_FIRMWARE_SECURE_ID]        = boot_platform_post_load_secure,
     [RSS_FIRMWARE_NON_SECURE_ID]    = boot_platform_post_load_non_secure,
     [RSS_FIRMWARE_SCP_ID]           = boot_platform_post_load_scp,
-    [RSS_FIRMWARE_LCP_ID]           = boot_platform_post_load_lcp,
     [RSS_FIRMWARE_SI_CL0_ID]        = boot_platform_post_load_si_cl0,
     [RSS_FIRMWARE_SI_CL1_ID]        = boot_platform_post_load_si_cl1,
     [RSS_FIRMWARE_SI_CL2_ID]        = boot_platform_post_load_si_cl2,
diff --git a/platform/ext/target/arm/rss/kronos/bl2/flash_map_bl2.c b/platform/ext/target/arm/rss/kronos/bl2/flash_map_bl2.c
index dc0ee9585..70bba5b6b 100644
--- a/platform/ext/target/arm/rss/kronos/bl2/flash_map_bl2.c
+++ b/platform/ext/target/arm/rss/kronos/bl2/flash_map_bl2.c
@@ -110,20 +110,6 @@ struct flash_area flash_map[] = {
         .fa_off = FLASH_AREA_13_OFFSET,
         .fa_size = FLASH_AREA_13_SIZE,
     },
-    {
-        .fa_id = FLASH_AREA_14_ID,
-        .fa_device_id = FLASH_DEVICE_ID,
-        .fa_driver = &FLASH_DEV_NAME,
-        .fa_off = FLASH_AREA_14_OFFSET,
-        .fa_size = FLASH_AREA_14_SIZE,
-    },
-    {
-        .fa_id = FLASH_AREA_15_ID,
-        .fa_device_id = FLASH_DEVICE_ID,
-        .fa_driver = &FLASH_DEV_NAME,
-        .fa_off = FLASH_AREA_15_OFFSET,
-        .fa_size = FLASH_AREA_15_SIZE,
-    },
     {
         .fa_id = AP_FLASH_AREA_0_ID,
         .fa_device_id = FLASH_DEVICE_ID,
@@ -160,10 +146,6 @@ int boot_get_image_exec_ram_info(uint32_t image_id,
         *exec_ram_start = HOST_SCP_IMG_BASE_S;
         *exec_ram_size  = HOST_SCP_ATU_SIZE;
         rc = 0;
-    } else if (image_id == RSS_FIRMWARE_LCP_ID) {
-        *exec_ram_start = HOST_LCP_IMG_BASE_S;
-        *exec_ram_size  = HOST_LCP_ATU_SIZE;
-        rc = 0;
     } else if (image_id == RSS_FIRMWARE_AP_BL2_ID) {
         *exec_ram_start = HOST_AP_BL2_IMG_BASE_S;
         *exec_ram_size  = HOST_AP_BL2_ATU_SIZE;
diff --git a/platform/ext/target/arm/rss/kronos/bl2/security_cnt_bl2.c b/platform/ext/target/arm/rss/kronos/bl2/security_cnt_bl2.c
index 9fa281d25..0b732d7c9 100644
--- a/platform/ext/target/arm/rss/kronos/bl2/security_cnt_bl2.c
+++ b/platform/ext/target/arm/rss/kronos/bl2/security_cnt_bl2.c
@@ -114,9 +114,6 @@ int32_t boot_nv_security_counter_update(uint32_t image_id,
             tfm_nv_counter_i = FWU_SI_CL0_NV_COUNTER;
             break;
         case PLAT_NV_COUNTER_BL2_6:
-            tfm_nv_counter_i = FWU_LCP_NV_COUNTER;
-            break;
-        case PLAT_NV_COUNTER_BL2_7:
             tfm_nv_counter_i = FWU_SCP_NV_COUNTER;
             break;
         default:
diff --git a/platform/ext/target/arm/rss/kronos/config.cmake b/platform/ext/target/arm/rss/kronos/config.cmake
index a8090653b..00dad9414 100644
--- a/platform/ext/target/arm/rss/kronos/config.cmake
+++ b/platform/ext/target/arm/rss/kronos/config.cmake
@@ -10,18 +10,11 @@ set(TFM_BL1_MEMORY_MAPPED_FLASH             OFF      CACHE BOOL    "Whether BL1
 set(PLATFORM_HAS_BOOT_DMA                   OFF      CACHE BOOL    "Enable dma support for memory transactions for bootloader")
 set(RSS_USE_HOST_UART                       OFF      CACHE BOOL    "Whether RSS should use the UART from the host system (opposed to dedicated UART)")
 set(RSS_USE_HOST_FLASH                      OFF      CACHE BOOL    "Enable RSS using the host flash.")
-set(MCUBOOT_IMAGE_NUMBER                    9        CACHE STRING  "Number of images supported by MCUBoot")
+set(MCUBOOT_IMAGE_NUMBER                    7        CACHE STRING  "Number of images supported by MCUBoot")
 set(ENABLE_SCP_ATU_CTRL                     OFF      CACHE BOOL    "Enable SCP to configure ATU")
 set(RSS_HAS_EXPANSION_PERIPHERALS           ON       CACHE BOOL    "Whether RSS has sub-platform specific peripherals in the expansion layer")
 set(PLATFORM_DEFAULT_MCUBOOT_IMAGE_NUMBER   OFF      CACHE BOOL    "Whether to use default value of MCUBOOT_IMAGE_NUMBER")
 set(ITS_MAX_ASSET_SIZE                      4096     CACHE STRING  "The maximum asset size to be stored in the Internal Trusted Storage")
-if (TFM_PLATFORM_VARIANT STREQUAL "1")
-    set(RSS_LCP_COUNT                          8        CACHE STRING   "Number of LCPs to load")
-elseif(TFM_PLATFORM_VARIANT STREQUAL "2")
-    set(RSS_LCP_COUNT                          4        CACHE STRING   "Number of LCPs to load")
-else()
-    set(RSS_LCP_COUNT                          16       CACHE STRING   "Number of LCPs to load")
-endif()
 
 # Maximum number of MCUBoot images supported by TF-M NV counters and ROTPKs
 if(MCUBOOT_IMAGE_NUMBER GREATER 9)
diff --git a/platform/ext/target/arm/rss/kronos/firmware_image_id.h b/platform/ext/target/arm/rss/kronos/firmware_image_id.h
index 5dcfbdfd2..583ab9296 100644
--- a/platform/ext/target/arm/rss/kronos/firmware_image_id.h
+++ b/platform/ext/target/arm/rss/kronos/firmware_image_id.h
@@ -21,9 +21,8 @@
 #define RSS_FIRMWARE_SI_CL2_ID       3
 #define RSS_FIRMWARE_SI_CL1_ID       4
 #define RSS_FIRMWARE_SI_CL0_ID       5
-#define RSS_FIRMWARE_LCP_ID          6
-#define RSS_FIRMWARE_SCP_ID          7
+#define RSS_FIRMWARE_SCP_ID          6
 /* Number of firmware loaded by BL2 */
-#define RSS_FIRMWARE_COUNT           8
+#define RSS_FIRMWARE_COUNT           7
 
 #endif /* __RSS_FIRMWARE_IMAGE_ID_H__ */
diff --git a/platform/ext/target/arm/rss/kronos/flash_layout.h b/platform/ext/target/arm/rss/kronos/flash_layout.h
index 0ceca3030..094422652 100644
--- a/platform/ext/target/arm/rss/kronos/flash_layout.h
+++ b/platform/ext/target/arm/rss/kronos/flash_layout.h
@@ -32,8 +32,6 @@
  * 0x3114_0000 Non-secure image secondary slot (384 KB)
  * 0x311A_0000 SCP BL1 primary slot (512 KB)
  * 0x3122_0000 SCP BL1 secondary slot (512 KB)
- * 0x313A_0000 LCP BL1 primary slot (64 KB)
- * 0x313B_0000 LCP BL1 secondary slot (64 KB)
  * 0x312A_0000 AP BL2 primary slot (512 KB)
  * 0x3132_0000 AP BL2 secondary slot (512 KB)
  */
@@ -51,7 +49,6 @@
 #define FLASH_S_PARTITION_SIZE          (SIZE_DEF_S_IMAGE)   /* S   partition */
 #define FLASH_NS_PARTITION_SIZE         (SIZE_DEF_NS_IMAGE)  /* NS  partition */
 #define FLASH_SCP_PARTITION_SIZE        (SIZE_DEF_SCP_IMAGE) /* SCP partition */
-#define FLASH_LCP_PARTITION_SIZE        (SIZE_DEF_LCP_IMAGE) /* LCP  partition */
 #define FLASH_AP_BL2_PARTITION_SIZE     (SIZE_DEF_AP_BL2_IMAGE) /* AP BL2  partition */
 #define FLASH_SI_CL0_PARTITION_SIZE     (SIZE_DEF_SI_CL0_IMAGE)  /* SI CL0 partition */
 #define FLASH_SI_CL1_PARTITION_SIZE     (SIZE_DEF_SI_CL1_IMAGE)  /* SI CL1 partition */
@@ -109,40 +106,32 @@
 #define FLASH_AREA_7_ID            (FLASH_AREA_6_ID + 1)
 #define FLASH_AREA_7_OFFSET        (FLASH_AREA_6_OFFSET + FLASH_AREA_6_SIZE)
 #define FLASH_AREA_7_SIZE          (FLASH_SCP_PARTITION_SIZE)
-/* LCP image primary slot */
+/* SI0 image primary slot */
 #define FLASH_AREA_8_ID            (FLASH_AREA_7_ID + 1)
 #define FLASH_AREA_8_OFFSET        (FLASH_AREA_7_OFFSET + FLASH_AREA_7_SIZE)
-#define FLASH_AREA_8_SIZE          (FLASH_LCP_PARTITION_SIZE)
-/* LCP image secondary slot */
+#define FLASH_AREA_8_SIZE          (FLASH_SI_CL0_PARTITION_SIZE)
+/* SI0 image secondary slot */
 #define FLASH_AREA_9_ID            (FLASH_AREA_8_ID + 1)
 #define FLASH_AREA_9_OFFSET        (FLASH_AREA_8_OFFSET + FLASH_AREA_8_SIZE)
-#define FLASH_AREA_9_SIZE          (FLASH_LCP_PARTITION_SIZE)
-/* SI0 image primary slot */
+#define FLASH_AREA_9_SIZE          (FLASH_SI_CL0_PARTITION_SIZE)
+/* SI1 image primary slot */
 #define FLASH_AREA_10_ID            (FLASH_AREA_9_ID + 1)
 #define FLASH_AREA_10_OFFSET        (FLASH_AREA_9_OFFSET + FLASH_AREA_9_SIZE)
-#define FLASH_AREA_10_SIZE          (FLASH_SI_CL0_PARTITION_SIZE)
-/* SI0 image secondary slot */
+#define FLASH_AREA_10_SIZE          (FLASH_SI_CL1_PARTITION_SIZE)
+/* SI1 image secondary slot */
 #define FLASH_AREA_11_ID            (FLASH_AREA_10_ID + 1)
 #define FLASH_AREA_11_OFFSET        (FLASH_AREA_10_OFFSET + FLASH_AREA_10_SIZE)
-#define FLASH_AREA_11_SIZE          (FLASH_SI_CL0_PARTITION_SIZE)
-/* SI1 image primary slot */
+#define FLASH_AREA_11_SIZE          (FLASH_SI_CL1_PARTITION_SIZE)
+/* SI2 image primary slot */
 #define FLASH_AREA_12_ID            (FLASH_AREA_11_ID + 1)
 #define FLASH_AREA_12_OFFSET        (FLASH_AREA_11_OFFSET + FLASH_AREA_11_SIZE)
-#define FLASH_AREA_12_SIZE          (FLASH_SI_CL1_PARTITION_SIZE)
-/* SI1 image secondary slot */
+#define FLASH_AREA_12_SIZE          (FLASH_SI_CL2_PARTITION_SIZE)
+/* SI2 image secondary slot */
 #define FLASH_AREA_13_ID            (FLASH_AREA_12_ID + 1)
 #define FLASH_AREA_13_OFFSET        (FLASH_AREA_12_OFFSET + FLASH_AREA_12_SIZE)
-#define FLASH_AREA_13_SIZE          (FLASH_SI_CL1_PARTITION_SIZE)
-/* SI2 image primary slot */
-#define FLASH_AREA_14_ID            (FLASH_AREA_13_ID + 1)
-#define FLASH_AREA_14_OFFSET        (FLASH_AREA_13_OFFSET + FLASH_AREA_13_SIZE)
-#define FLASH_AREA_14_SIZE          (FLASH_SI_CL2_PARTITION_SIZE)
-/* SI2 image secondary slot */
-#define FLASH_AREA_15_ID            (FLASH_AREA_14_ID + 1)
-#define FLASH_AREA_15_OFFSET        (FLASH_AREA_14_OFFSET + FLASH_AREA_14_SIZE)
-#define FLASH_AREA_15_SIZE          (FLASH_SI_CL2_PARTITION_SIZE)
+#define FLASH_AREA_13_SIZE          (FLASH_SI_CL2_PARTITION_SIZE)
 /* FIP image primary slot, in AP secure flash */
-#define AP_FLASH_AREA_0_ID          (FLASH_AREA_15_ID + 1)
+#define AP_FLASH_AREA_0_ID          (FLASH_AREA_13_ID + 1)
 #define AP_FLASH_AREA_0_OFFSET      (0)
 #define AP_FLASH_AREA_0_SIZE        (FLASH_AP_BL2_PARTITION_SIZE)
 /* FIP image secondary slot, in AP secure flash */
@@ -163,20 +152,18 @@
         (((x) == RSS_FIRMWARE_SECURE_ID)     ? FLASH_AREA_2_ID : \
          ((x) == RSS_FIRMWARE_NON_SECURE_ID) ? FLASH_AREA_3_ID : \
          ((x) == RSS_FIRMWARE_SCP_ID)        ? FLASH_AREA_6_ID : \
-         ((x) == RSS_FIRMWARE_LCP_ID)        ? FLASH_AREA_8_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL0_ID)     ? FLASH_AREA_10_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL1_ID)     ? FLASH_AREA_12_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL2_ID)     ? FLASH_AREA_14_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL0_ID)     ? FLASH_AREA_8_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL1_ID)     ? FLASH_AREA_10_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL2_ID)     ? FLASH_AREA_12_ID : \
          ((x) == RSS_FIRMWARE_AP_BL2_ID)     ? AP_FLASH_AREA_0_ID : \
                                               255)
 #define FLASH_AREA_IMAGE_SECONDARY(x) \
         (((x) == RSS_FIRMWARE_SECURE_ID)     ? FLASH_AREA_4_ID : \
          ((x) == RSS_FIRMWARE_NON_SECURE_ID) ? FLASH_AREA_5_ID : \
          ((x) == RSS_FIRMWARE_SCP_ID)        ? FLASH_AREA_7_ID : \
-         ((x) == RSS_FIRMWARE_LCP_ID)        ? FLASH_AREA_9_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL0_ID)     ? FLASH_AREA_11_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL1_ID)     ? FLASH_AREA_13_ID : \
-         ((x) == RSS_FIRMWARE_SI_CL2_ID)     ? FLASH_AREA_15_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL0_ID)     ? FLASH_AREA_9_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL1_ID)     ? FLASH_AREA_11_ID : \
+         ((x) == RSS_FIRMWARE_SI_CL2_ID)     ? FLASH_AREA_13_ID : \
          ((x) == RSS_FIRMWARE_AP_BL2_ID)     ? AP_FLASH_AREA_1_ID : \
                                               255)
 
diff --git a/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.c b/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.c
index fed777078..9bbe38ee8 100644
--- a/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.c
+++ b/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.c
@@ -651,11 +651,8 @@ static enum fwu_agent_error_t update_nv_counters(
         case FWU_SI_CL0_NV_COUNTER:
             tfm_nv_counter_i = PLAT_NV_COUNTER_BL2_5;
             break;
-        case FWU_LCP_NV_COUNTER:
-            tfm_nv_counter_i = PLAT_NV_COUNTER_BL2_6;
-            break;
         case FWU_SCP_NV_COUNTER:
-            tfm_nv_counter_i = PLAT_NV_COUNTER_BL2_7;
+            tfm_nv_counter_i = PLAT_NV_COUNTER_BL2_6;
             break;
         default:
             FWU_ASSERT(0);
diff --git a/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.h b/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.h
index 5b13a9a0c..caadb48e4 100644
--- a/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.h
+++ b/platform/ext/target/arm/rss/kronos/fw_update_agent/fwu_agent.h
@@ -173,7 +173,6 @@ enum fwu_nv_counter_index_t {
     FWU_SI_CL2_NV_COUNTER,
     FWU_SI_CL1_NV_COUNTER,
     FWU_SI_CL0_NV_COUNTER,
-    FWU_LCP_NV_COUNTER,
     FWU_SCP_NV_COUNTER,
     FWU_MAX_NV_COUNTER_INDEX = FWU_SCP_NV_COUNTER
 };
diff --git a/platform/ext/target/arm/rss/kronos/host_base_address.h b/platform/ext/target/arm/rss/kronos/host_base_address.h
index e9465a4a9..1fc6c75e1 100644
--- a/platform/ext/target/arm/rss/kronos/host_base_address.h
+++ b/platform/ext/target/arm/rss/kronos/host_base_address.h
@@ -100,9 +100,6 @@
  * Note: Some overlap is expected between ATU CODE and ATU HEADER in physical
  *       memory as the HEADER is smaller than the ATU page size. This is fine
  *       as the whole image is shifted and would have fit inside ATU CODE.
- * Note: The physical memory for MCP is used for all images which allows the
- *       code on other firmware (particularly LCP) to be large as the memory
- *       no longer needs to contain the header.
  */
 
 
@@ -129,34 +126,8 @@
 /* ATU ID for SCP ATU CODE region */
 #define HOST_SCP_IMG_CODE_ATU_ID    RSS_ATU_IMG_CODE_LOAD_ID
 
-/*
- * All LCP use the same ATU region but map it to a different physical address
- * The physical address is the cluster utility space in SCP which has an
- * address translation window for each LCP.
- * Note: The cluster utility space requires SCP's ATU to be setup correctly.
- */
-
-/* LCP ATU HEADER logical address start */
-#define HOST_LCP_HDR_ATU_BASE_S     (HOST_SCP_CODE_BASE_S + HOST_SCP_ATU_SIZE)
-/* LCP Code region and LCP ATU CODE logical address start */
-#define HOST_LCP_CODE_BASE_S        (HOST_LCP_HDR_ATU_BASE_S + RSS_IMG_HDR_ATU_SIZE)
-/* LCP Image address start, offset so end of HEADER is at end of ATU HEADER */
-#define HOST_LCP_IMG_BASE_S         (HOST_LCP_CODE_BASE_S - BL2_HEADER_SIZE)
-/* LCP0 ITCM window in SCP address start */
-#define HOST_LCP_0_PHYS_BASE        0x0000200050000ULL
-/* Offset between LCP ITCM windows in SCP */
-#define HOST_LCP_N_PHYS_OFFSET      0x200000U
-/* LCP ITCM window in SCP address start for LCP N*/
-#define HOST_LCP_N_PHYS_BASE(n)     (HOST_LCP_0_PHYS_BASE + HOST_LCP_N_PHYS_OFFSET * n)
-/* LCP ATU CODE size (aligned size of LCP image) */
-#define HOST_LCP_ATU_SIZE           ALIGN_UP(SIZE_DEF_LCP_IMAGE, RSS_ATU_PAGE_SIZE)
-/* ATU ID for LCP ATU HEADER region */
-#define HOST_LCP_IMG_HDR_ATU_ID     RSS_ATU_IMG_HDR_LOAD_ID
-/* ATU ID for LCP ATU CODE region */
-#define HOST_LCP_IMG_CODE_ATU_ID    RSS_ATU_IMG_CODE_LOAD_ID
-
 /* AP BL2 ATU HEADER logical address start */
-#define HOST_AP_BL2_HDR_ATU_BASE_S     (HOST_LCP_CODE_BASE_S + HOST_LCP_ATU_SIZE)
+#define HOST_AP_BL2_HDR_ATU_BASE_S     (HOST_SCP_CODE_BASE_S + HOST_SCP_ATU_SIZE)
 /*  AP BL2 Code region and SCP ATU CODE logical address start */
 #define HOST_AP_BL2_CODE_BASE_S        (HOST_AP_BL2_HDR_ATU_BASE_S + RSS_IMG_HDR_ATU_SIZE)
 /*  AP BL2 Image address start, offset so end of HEADER is at end of ATU HEADER */
diff --git a/platform/ext/target/arm/rss/kronos/size_defs.h b/platform/ext/target/arm/rss/kronos/size_defs.h
index d96ced152..441713df9 100644
--- a/platform/ext/target/arm/rss/kronos/size_defs.h
+++ b/platform/ext/target/arm/rss/kronos/size_defs.h
@@ -24,7 +24,6 @@
 
 /* Host Firmware */
 #define SIZE_DEF_SCP_IMAGE          (0x80000) /* 512 KB */
-#define SIZE_DEF_LCP_IMAGE          (0x10000) /* 64 KB */
 #define SIZE_DEF_AP_BL2_IMAGE       (0x80000) /* 512 KB */
 #define SIZE_DEF_SI_CL0_IMAGE       (0x200000) /* 2MB */
 #define SIZE_DEF_SI_CL1_IMAGE       (0x400000) /* 4MB */
-- 
2.25.1

